src/metabase/driver/drill.clj

```diff
@@ -0,0 +1,184 @@
+                                     1)
+                               3)
+                         (hsql/raw "INTERVAL '3' MONTH")))
+    ;; quarter gives incorrect results in Drill 1.10
+    ;;:quarter (hsql/call :date_trunc_quarter (hx/->timestamp expr))
+    :quarter-of-year (hx/->integer
+                      (hsql/call :ceil
+                                 (hx// (hsql/call :extract
+                                                  "month"
+                                                  (hx/->timestamp expr))
+                                       3.0)))
+    :year (hsql/call :extract "year" (hx/->timestamp expr))))
+
+(defmethod hformat/fn-handler "drill-from-unixtime" [_ datetime-literal]
+  (hformat/to-sql
+   (hsql/call :to_timestamp
+              ;;(hx/literal (u/date->iso-8601 datetime-literal))
+              datetime-literal
+              (hx/literal "YYYY-MM-dd''T''HH:mm:ss.SSSZ"))))
+
+(defn drill-unprepare
+  "Translates `sql-and-args` to the Drill SQL dialect"
+  [sql-and-args]
+  (unprepare/unprepare sql-and-args :iso-8601-fn :drill-from-unixtime))
+
+(defn- execute-query
+  "Process and run a native (raw SQL) QUERY."
+  [driver {:keys [database settings], query :native, :as outer-query}]
+  (let [query (-> (assoc query :remark (qputil/query->remark outer-query))
+                  (assoc :query (if (seq (:params query))
+                                  (drill-unprepare (cons (:query query) (:params query)))
+                                  (:query query)))
+                  (dissoc :params))]
+    (qp/do-with-try-catch
+     (fn []
+       (let [db-connection (sql/db->jdbc-connection-spec database)]
+         (hive-like/run-query-without-timezone driver settings db-connection query))))))
+
+;; This provides an implementation of `prepare-value` that prevents HoneySQL from converting forms to prepared statement parameters (`?`)
+;; TODO - Move this into `metabase.driver.generic-sql` and document it as an alternate implementation for `prepare-value` (?)
+;;        Or perhaps investigate a lower-level way to disable the functionality in HoneySQL, perhaps by swapping out a function somewhere
+(defprotocol ^:private IPrepareValue
+  (^:private prepare-value [this]))
+(extend-protocol IPrepareValue
+  nil (prepare-value [_] nil)
+  DateTimeValue (prepare-value [{:keys [value]}] (prepare-value value))
+  Value (prepare-value [{:keys [value]}] (prepare-value value))
+  String (prepare-value [this] (hx/literal this))
+  Boolean (prepare-value [this] (hsql/raw (if this "TRUE" "FALSE")))
+  Date (prepare-value [this] (hsql/call :to_timestamp
+                                        (hx/literal (u/date->iso-8601 this))
+                                        (hx/literal "YYYY-MM-dd''T''HH:mm:ss.SSSZ")))
+  Number (prepare-value [this] this)
+  Object (prepare-value [this] (throw (Exception. (format "Don't know how to prepare value %s %s" (class this) this)))))
```

Comment:
Copying this entire protocol and set of implementations over from the BigQuery to only change one line sounds like a like of code duplication. I'm happy to fix this and rework this in a way that avoids copy pasta but it's something I want to address before merging